### ISSUE TITLE
CustomSelectControl: currently selected value is not presented to screen readers

### ISSUE THREAD

**Issue by Aljullu** [WordPress/gutenberg#21459]
**Describe the bug**
The `CustomSelectControl` component doesn't read the currently selected value when it's focused.

**To reproduce**
1. Edit a post and select a paragraph block.
2. With a screen reader navigate the block _Typography_ panel in the sidebar.
3. Focus the _Preset Size_ select.
4. Notice the currently selected option is not presented to the user.

**Expected behavior**
When focusing the select, I would expect the currently selected value to be read by the screen reader. Like it's done for other form elements.

**Desktop (please complete the following information):**
Tested in two devices:

**Device 1**
 - OS: Fedora Linux
 - Browser: Firefox
 - Screen reader: Orca

**Device 2**
 - OS: Windows
 - Browser: Chrome
 - Screen reader: Narrator

**Comment by Aljullu** [comment]
This was closed in https://github.com/WordPress/gutenberg/pull/33941.
